lib/dnsmadeeasy-rest-api.rb

```diff
@@ -59,15 +59,15 @@ def records_for(domain_name)
   end
 
   def find(domain_name, name, type)
-    records = records_for(domain_name)
-    records['data'].detect { |r| r['name'] == name && r['type'] == type }
+    records = get "/dns/managed/#{get_id_by_domain(domain_name)}/records?recordName=#{name}&type=#{type}"
+
+    records['data'].first
   end
 
   def find_record_id(domain_name, name, type)
-    records = records_for(domain_name)
+    records = get "/dns/managed/#{get_id_by_domain(domain_name)}/records?recordName=#{name}&type=#{type}"
```

Comment:
Think it might be worth extracting this call since it's shared between the two here.